### PR TITLE
prov/rxd: added counter support

### PR DESCRIPTION
**Pull request by hoopoepg**
- added counter support to RXD provider

Signed-off-by: Oblomov, Sergey <sergey.oblomov@intel.com>

### CODE REVIEW

**Comment by hoopoepg** [comment]
this PR includes https://github.com/ofiwg/libfabric/pull/3158 (to allow success compilation)

File: prov/rxd/src/rxd_cntr.c [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+	struct util_cntr *cntr;
+
+	switch (rx_entry->op_hdr.op) {
+	case ofi_op_msg:
+	case ofi_op_tagged:
+		cntr = ep->util_ep.rx_cntr;
+		break;
+	case ofi_op_atomic:
+	case ofi_op_write:
+		cntr = ep->util_ep.wr_cntr;
```
Comment (shefty) [inline prov/rxd/src/rxd_cntr.c]: rem_wr_cntr?

File: prov/rxd/src/rxd_cntr.c [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+void rxd_cntr_report_error(struct rxd_ep *ep, struct fi_cq_err_entry *err)
+{
+        struct util_cntr *cntr;
+
+	cntr = RXD_FLAG(err->flags, (FI_WRITE)) ? ep->util_ep.wr_cntr :
+	       RXD_FLAG(err->flags, (FI_ATOMIC)) ? ep->util_ep.wr_cntr :
+	       RXD_FLAG(err->flags, (FI_READ)) ? ep->util_ep.rd_cntr :
+	       RXD_FLAG(err->flags, (FI_SEND)) ? ep->util_ep.tx_cntr :
+	       RXD_FLAG(err->flags, (FI_RECV)) ? ep->util_ep.rx_cntr :
+	       0;
```
Comment (shefty) [inline prov/rxd/src/rxd_cntr.c]: 0 -> NULL

**Comment by shefty** [comment]
PR has new conflicts now.  Can you please resolve and re-push?

File: prov/rxd/src/rxd_cntr.c (hunk 1 above)
Comment (hoopoepg) [inline prov/rxd/src/rxd_cntr.c]: yep, wrong counter :)
updated

File: prov/rxd/src/rxd_cntr.c (hunk 2 above)
Comment (hoopoepg) [inline prov/rxd/src/rxd_cntr.c]: fixed
thank you

**Comment by hoopoepg** [comment]
PR updated (conflicts are resolved)

**Pull request merged**